packages/rum/src/viewCollection.ts

```diff
@@ -105,11 +105,18 @@ function newView(
   const { stop: stopTimingsTracking } = trackTimings(lifeCycle, updateMeasures)
   const { stop: stopEventCountsTracking } = trackEventCounts(lifeCycle, updateMeasures)
 
-  function updateLoadingTime(loadingTimeValue: number | undefined) {
-    loadingTime = loadingTimeValue
-    scheduleViewUpdate()
+  function updateLoadingTime(loadingTimeValue: number) {
+    if (loadingTime === undefined || loadingTimeValue > loadingTime) {
```

Comment:
IMO, it would be easier to maintain if all the logic related to loading time is encapsulated. 
It would also be easier to understand the intent by expressing that:
- we are waiting several timings
- we want the max of them
It would avoid to send an update with an incorrect or changing loadingTime value .

We could then have something like:

```
{stop: stopLoadingTime} = trackLoadingTime(lifecycle, loadingType, callback) {
  let expectedTiming = 1
  const receivedTimings = []
  if (initial view) {
    expectedTiming++
    // load event logic
  }

  // activity logic

  function onTimingValue(timingValue) {
    receivedTimings.push(timingValue)
    expectedTiming--
    if (expectedTiming === 0) {
      callback(Math.max(receivedTimings))
    }
  }

  return {
    stop() {
      // stop load event and activity
    }
  }
}
```

What do you think?

Reply:
It would be easier to understand for sure. I will use this structure to encapsulate the two `trackLoadEventLoadingTime` and `trackActivityLoadingTime` calls as it will also simplify the implementation of your approach.